src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -2184,20 +2184,13 @@ private void OnThreadException(Exception e)
                             continue;
                         }
 
-                        submission.CompleteLogging(false);
```

Comment:
Note that this call could have completed the submission and mutating the build result a few lines below could have run in parallel with the completion handler and cause grief.